tests/PidTest.cpp

```diff
@@ -0,0 +1,331 @@
+/**
+ * @file PidTest.cpp
+ * @brief PidGainクラス, Pidクラスをテストする
+ * @author miyahara046
+ */
+
+#include "Pid.h"
+#include "gtest/gtest.h"
+namespace etrobocon2025_test {
+
+  // PidGainの正のゲインがそのまま格納されるかをテスト
+  TEST(PidGainTest, PidGainPlus)
+  {
+    double kp = 0.1;
+    double ki = 0.03;
+    double kd = 0.612;
+    double expected_kp = 0.1;
+    double expected_ki = 0.03;
+    double expected_kd = 0.612;
+    PidGain actualPidGain(kp, ki, kd);
+    EXPECT_DOUBLE_EQ(expected_kp, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected_ki, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected_kd, actualPidGain.kd);
+  }
+  // PidGainの0のゲインがそのまま格納されるかをテスト
+  TEST(PidGainTest, PidGainZero)
+  {
+    double kp = 0.0;
+    double ki = 0.0;
+    double kd = 0.0;
+    double expected_kp = 0.0;
+    double expected_ki = 0.0;
+    double expected_kd = 0.0;
+    PidGain actualPidGain(kp, ki, kd);
+    EXPECT_DOUBLE_EQ(expected_kp, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected_ki, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected_kd, actualPidGain.kd);
+  }
+
+  // PidGainの負のゲインが0に補正されるかをテスト
+  TEST(PidGainTest, PidGainMinus)
+  {
+    double kp = -0.5;
+    double ki = -0.2;
+    double kd = -0.3;
+    PidGain actualPidGain(kp, ki, kd);
+    double expected = 0.0;
+    EXPECT_DOUBLE_EQ(expected, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected, actualPidGain.kd);
+  }
+
+  // 基本的なPID計算の結果が正しいかをテスト
+  TEST(PidTest, CalculatePid)
+  {
+    Pid actualPid(0.6, 0.02, 0.03, 70, 100.0, -100.0);
+    double currentValue = 20;
+    /** 計算過程
+     * 1. 前回の誤差
+     * prevDeviation = 0
+     * 2. 現在の誤差
+     * currentDeviation = (70 - 20) = 50
+     * 3. 誤差の積分を計算
+     * integral = 0 + (50 + 0) * 0.01 / 2 = 0.25
+     * 4. 微分の処理を行う
+     * currentDerivative = (50 - 0) / 0.01 = 5000
+     * 微分項にローパスフィルタを適用
+     * prevDeviat filteredDerivative = 0.8 * 5000 + (1 - 0.8) * 0 = 4000
+     * 5. PID制御を計算
```

Comment:
```suggestion
     * 5. 微分項にローパスフィルタを適用
     * prevDeviat filteredDerivative = 0.8 * 5000 + (1 - 0.8) * 0 = 4000
     * 6. PID制御を計算
```
計算過程めっちゃ頑張ってるから、これ出すのも申し訳ないんだが、番号が途中抜けてるかな。
他のところも。